client/src/main/java/org/apache/celeborn/client/read/DfsPartitionReader.java

```diff
@@ -122,7 +126,7 @@ public DfsPartitionReader(
           e);
     }
 
-    if (endMapIndex != Integer.MAX_VALUE) {
+    if (endMapIndex != Integer.MAX_VALUE && endMapIndex != -1) {
```

Comment:
If the feature of optimizing skew partition read has been enabled, there will be no more sort. So here should not read the sorted DFS file.
The condition should be the condition startMapIndex > endMapIndex to be true and read chunk range from DFS shuffle file.

Reply:
In my understanding, currently celeborn supports both sort-based skew-handling and the sort-free skew-handling introduced in https://github.com/apache/celeborn/pull/2373. If we change the condition to `startMapIndex > endMapIndex`, then it will no longer support sort-based skew-handling at all.
There are some other sort-based skew-handling code in current codebase, such as `PartitionFilesSorter`. If we want to completely remove the code related to sort-based skew-handling, would it be more appropriate to do it in a new PR after https://github.com/apache/celeborn/pull/3118 is merged into the main branch?
Please correct me if there's anything I haven't understood correctly.